report.php

```diff
@@ -319,11 +319,8 @@
     $saveallbutton .= '<input type="submit" class="btn btn-primary" style="border-radius: 8px" value="'
                       .get_string("saveallfeedback", "diary").'" />';
     // 20200421 Added a return button.
-    $url = new moodle_url('mod/diary/view.php', ['id' => $id]);
-    $saveallbutton .= ' <a href="'.$url->out(false)
-                     .'" class="btn btn-secondary" role="button" style="border-radius: 8px">'
-                     .get_string('returnto', 'diary', $diary->name)
-                     .'</a>';
+    $url = new moodle_url('/mod/diary/view.php', ['id' => $id]);
+    $saveallbutton .= html_writer::link($url, get_string('returnto', 'diary', $diary->name), ['class' => 'btn btn-secondary', 'role' => 'button', 'style' => 'border-radius: 8px']);
```

Comment:
does use moodle_url and use html_writer need to be added to the top of this page ?